core/src/main/java/com/sequenceiq/cloudbreak/reactor/handler/cluster/ClusterDbCertRotationHandler.java

```diff
@@ -31,7 +48,10 @@ protected Selectable defaultFailureEvent(Long resourceId, Exception e, Event<Clu
     protected Selectable doAccept(HandlerEvent<ClusterDbCertRotationRequest> event) {
         LOGGER.info("Start rotation of the DB cert on the Cluster");
         ClusterDbCertRotationRequest request = event.getData();
-        LOGGER.info("Skeleton of the cert rotation of the DB cert");
+        Long resourceId = event.getData().getResourceId();
+        StackDto stackDto = stackDtoService.getById(resourceId);
+        Set<String> rootSslCerts = dbCertificateProvider.getRelatedSslCerts(stackDto).getSslCerts();
+        clusterService.updateDbSslCert(stackDto.getCluster().getId(), rootSslCerts);
```

Comment:
If `rootSslCerts.isEmpty()`, there is no need to update the fields in `Cluster`. It is not possible to have this set non-empty at launch and cleared later; the cert rotation will always result in a non-empty root cert set (if the SSL is enabled). So I suggest adding an `if` around the `updateDbSslCert()` call.